pkg/network/vmispec/interface.go

```diff
@@ -58,6 +58,24 @@ func IsPodNetworkWithMasqueradeBindingInterface(networks []v1.Network, ifaces []
 	return true
 }
 
+func ShiftPrimaryInterface(statusIfaces []v1.VirtualMachineInstanceNetworkInterface, networks []v1.Network) []v1.VirtualMachineInstanceNetworkInterface {
```

Comment:
The name is not telling me to where it is shift.
There are several ways to adjust it, some reveals the fact it does too many things and some breaks it into pieces.

I would go with something like this:
```
If primaryIface := PopInterfaceByNetwork(statusIfaces, lookupPodNetwork(networks)): primaryInterface =! nil {
    statusIfaces = append([]v1.stuff{primaryIface}, statusIfaces…)
}
```

All this at the caller.

Reply:
done